### PR TITLE
Revert "Revert "Mk nikshay audit cookie safe""

### PR DESCRIPTION
**Pull request by mkangia**
Reverts dimagi/commcare-hq#16005

This fixes the issue introduced due to changes in app preview.
Adding hold off on merge to add a notification in case csrftoken is missed anywhere to better tracking.

### CODE REVIEW

File: corehq/apps/hqwebapp/static/hqwebapp/js/ajax_csrf_setup.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ $.ajaxSetup({
     beforeSend: function(xhr, settings) {
         // Don't pass csrftoken cross domain
         if (!csrfSafeMethod(settings.type) && !this.crossDomain) {
-            $csrf_token = $.cookie('csrftoken');
+            $csrf_token = $("#csrfTokenContainer").val();
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/static/hqwebapp/js/ajax_csrf_setup.js]: '$csrf_token' is not defined. (no-undef)

File: corehq/apps/hqwebapp/middleware.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        warning = "{ajax}Request at {url} doesn't contain a csrf token. Referring url is {referer}."\
+                  "Possibly related to https://github.com/dimagi/commcare-hq/pull/16008".format(
+                      url=request_url,
+                      referer=referring_url,
+                      ajax=('Ajax ' if request.is_ajax() else '')
+                  )
+        _assert = soft_assert('{}@{}'.format('mkangia+logs', 'dimagi.com'), exponential_backoff=False)
+        _assert(False, warning)
+
+        return super(HQCsrfViewMiddleWare, self)._reject(request, reason)
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/middleware.py]: W292 no newline at end of file

**Comment by mkangia** [comment]
@sravfeyn thanks for the suggestion for notification. Added!
@orangejenny added csrftoken missed earlier in app preview base view.

**Comment by orangejenny** [comment]
Code looks fine to me, but csrf test is failing: `AssertionError: Request at /a/delhi/sms/send_to_recipients/ doesn't contain a csrf token.`

**Comment by mkangia** [comment]
Aha! fixed it. Thanks for flagging @orangejenny

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/hqwebapp/middleware.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/middleware.py]: W391 blank line at end of file

**Pull request merged**